### ISSUE TITLE
Flutter Markdown should pass through onSelectionChanged of SelectableText

### ISSUE THREAD

**Issue by yorickreum** [flutter/flutter#128170]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

This issue was already raised in [#81715](https://github.com/flutter/flutter/issues/81715) and is probably related to #118045 as well.

It is currently impossible to figure out which text was selected by a user in the Flutter Markdown widget if ```selectable = true```. This is quite annoying especially since it is a fundamental feature of ```SelectableText```.

### Proposal

The easiest fix would be to just pass through the ```onSelectionChanged``` callback from ```SelectableText.rich```. Preferably of course also the other callbacks.

**Comment by yorickreum** [comment]
Alternatively, drop ```selectable``` and implement https://github.com/flutter/flutter/issues/107073 instead.

**Comment by huycozy** [comment]
Yes, I also think this can be achieved with https://github.com/flutter/flutter/issues/107073 but I'm not sure if `SelectableText` will be deprecated or not. Keeping this issue open for further information.

**Comment by masus04** [comment]
onSelectionChanged is now supported, however, there is no way to change the selection from within the method, as far as I can tell. It would be nice if the callback method would expect a `Selection?` return value, which would be used as the new selection.
